### ISSUE TITLE
Shift-space-swipe selection should also work when pressing keys that replace Shift (`=\<`, `?123`)

### ISSUE THREAD

**Issue by klaurence** [florisboard/florisboard#1998]
When pressing and holding the `=\<` and `?123` _when they have replaced the regular Shift **`^`** key_ (which happens at the primary and secondary symbol pages) and then swiping the space bar to select text, text should be selected (like when holding the regular Shift key).

<details>
<summary><i>Initially-filed bug info</i></summary>

#### Steps to reproduce

1. In a text box, type some example text.
2. Click the `?123` key below the regular Shift key to go to the primary symbols page.
3. Place the cursor at the middle of the example text.
4. Press and gold the `=\<` key that has replaced the regular Shift key and swipe the spacebar to the left and the right.
5. (The behavior should be the same after clicking `=\<`, holding the `?123` key that has replaced the `=\<` key, and then swiping left and right.)

#### Environment information

- FlorisBoard Version: 0.4.0-alpha01
- Android: 10
</details>

**Comment by tsiflimagas** [comment]
This doesn't seem like a bug to me, as I don't think it was ever meant to work that way. `=\<` maybe be in the same position as the shift key, but it doesn't replicate its functionality, and probably doesn't have to. Unless I'm wrong, this should be converted to a proposal.

**Comment by patrickgold** [comment]
I agree with @tsiflimagas here, not a bug, however I somewhat understand where the idea and the initial assumption comes from. The problem is that on PC keyboards, one uses the shift key to switch from the first layer to the second layer. This applies to the letters, but **also** for the symbols. So one could assume that this is also true for soft keyboards, which it isn't. FlorisBoard does not really think in layers, but sees the character, symbol and symbol2 "layers" as completely different **layouts**. Thus the `=\<` key is in fact a layout toggle key instead of a shift key and also behaves that way. The upper- and lowercase characters you see in the keyboard are in fact also not layers, but just the same key instance, just with dynamically adjusting data based on the shift key state.

Integrating the `=\<` key into the shift logic so shift+space works is an interesting and good proposal though, but I am not sure if this is possible without severly messing up the symbols and layout logic. Will properly relabel this issue.

**Comment by KLaurence** [comment]
> This doesn't seem like a bug to me, as I don't think it was ever meant to work that way. `=\<` maybe be in the same position as the shift key, but it doesn't replicate its functionality, and probably doesn't have to. Unless I'm wrong, this should be converted to a proposal.

> I agree with @<>tsiflimagas here, not a bug [...] Integrating the `=\<` key into the shift logic so shift+space works is an interesting and good proposal though, but I am not sure if this is possible without severely messing up the symbols and layout logic.

I initially thought that it was unintentional that the spacebar selection doesn't work for the `=\<` and `?123` keys (the keys that replace Shift's position) which made me file it as a bug. My idea was more or less "this key allows me to select text by swiping the spacebar, so I think these keys which replace that key should've also allowed me to select text for consistency, but developers might've forgotten to add it".

Since the issue was recategorized as a proposal, I'll reformat the title and original-comment contents to better fit as proposal.
